src/mongocxx/database.cpp

```diff
@@ -157,5 +164,12 @@ collection database::collection(stdx::string_view name) const {
     return mongocxx::collection(*this, name);
 }
 
+const std::unique_ptr<database::impl>& database::get_impl() const {
+    if(!_impl) {
+        throw operation_exception{make_error_code(error_code::k_database_uninitialized)};
```

Comment:
operation_exception is fine with me.  I'm not sure what you mean by the second question.